src/dvc_data/hashfile/transfer.py

```diff
@@ -89,6 +89,10 @@ def _do_transfer(
         bound_file_ids = all_file_ids & entry_ids
         all_file_ids -= entry_ids
 
+        logger.trace(
+            f"transfer dir: {dir_hash} with {len(bound_file_ids)} files"
```

Comment:
Probably doesn't matter here, but using f strings in logging is not desirable
```suggestion
            "transfer dir: %s with %d files", dir_hash, len(bound_file_ids)
```

Reply:
Interesting, I didn't know there was a speed difference between the format method and the percent operator. But it's real:

![image](https://github.com/iterative/dvc-data/assets/3186211/c2742e08-1fd9-4716-9555-31f1c70f4b3d)

Code for above benchmark: 
https://github.com/Erotemic/timerit/blob/dev/1.0.2/examples/benchmark_string_formatting.py

Reply:
Although intrestingly, if you look at f-strings themselves, they seem to perform better than the format method (at least on Python 3.11). However, I'm not sure if my methodology for testing f-strings is correct. In order to vary the number of arguments and format types I had to use some `eval` hacking to define a function with an appropriate f-string for each case that I wanted to test. This could skew the results, but if it doesn't then when the number of variables you are formatting is 2 or less, then f-strings seem to have a small advantage for simple strings / integers. 

![image](https://github.com/iterative/dvc-data/assets/3186211/cdb87a9f-dbd5-42cb-8516-156c42c8588f)

The above plot is messy so here is one that restricts only to the `%d` and `%s` cases:

![Uploading image.png…]()

Reply:
@Erotemic It is really about lazy logging rather than % vs f strings. In this particular case it is likely negligible, but we've been burned before and try to keep it consistent so we don't have to guess.